src/popup/components/Popup.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+import { runtime } from 'webextension-polyfill';
+import { fetchUser } from '../../gkApi';
+import type { PermissionsRequest } from '../../permissions-helper';
+import { PopupInitMessage } from '../../shared';
+import type { User } from '../../types';
+import { RequestPermissionsMenuItem } from './RequestPermissionsMenuItem';
+import { SignedInMenuItems } from './SignedInMenuItems';
+import { SignedOutMenuItems } from './SignedOutMenuItems';
+import { SupportMenuItem } from './SupportMenuItem';
+
+const syncWithBackground = async () => {
+	return (await runtime.sendMessage(PopupInitMessage)) as PermissionsRequest | undefined;
+};
+
+export const Popup = () => {
+	const [permissionsRequest, setPermissionsRequest] = useState<PermissionsRequest | undefined>(undefined);
+	const [user, setUser] = useState<User | null>(null);
+	const [isLoading, setIsLoading] = useState(true);
+
+	useEffect(() => {
+		const loadData = async () => {
+			const permissionsRequest = await syncWithBackground();
```

Comment:
nit: The logic looks correct, but this can be confusing because it masks `permissionsRequest` from `useState` within this block.